src/services/messengerSync/producer.js

```diff
@@ -0,0 +1,80 @@
+const { Configuration } = require('@schul-cloud/commons');
+const { createChannel } = require('../../utils/rabbitmq');
+
+const QUEUE_INTERNAL = Configuration.get('RABBITMQ_MATRIX_QUEUE_INTERNAL');
+
+let app;
+let channel;
+
+const sendToQueue = (message) => {
+	const msgJson = JSON.stringify(message);
+	const msgBuffer = Buffer.from(msgJson);
+	channel.sendToQueue(QUEUE_INTERNAL, msgBuffer, { persistent: true });
+};
+
+const requestFullSchoolSync = (school) => {
+	const message = {
+		action: 'syncSchool',
+		schoolId: school._id,
+		fullSync: true,
+	};
+	sendToQueue(message);
+};
+
+const requestSyncForEachCourseUser = async (course) => {
+	const users = course.userIds.concat(course.teacherIds).concat(course.substitutionIds);
```

Comment:
this is BL: getAllCourseUsers(courseOrCourseId) and should be testable extracted to smth like usergroup/logic/courses and reused, at least update const name
```suggestion
	const courseUserIds = course.userIds.concat(course.teacherIds).concat(course.substitutionIds);
```